### DISCUSSION TITLE
Getting Pull request details using commit sha

### DISCUSSION THREAD

**Discussion by subodh2711** [community/community#22481] [Pull Requests]
<p>I have all the commit details , how can i get the pull request details corresponding to that commit SHA or commit url?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>What exactly are you trying to achieve? In general commits exist independent of PRs, while PRs contain a set of commits to merge (which may change if the PR is updated).</p>

**Comment by jsoref** [comment]
<p>I imagine you’re trying to reproduce the equivalent of:</p><aside class="onebox githubcommit" data-onebox-src="https://github.com/github/docs/commit/8e25c7666ce5b9576d6aa1e982a58285f5f90ef9">
  <header class="source">

      <a href="https://github.com/github/docs/commit/8e25c7666ce5b9576d6aa1e982a58285f5f90ef9" target="_blank" rel="noopener nofollow ugc">github.com/github/docs</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M10.86 7c-.45-1.72-2-3-3.86-3-1.86 0-3.41 1.28-3.86 3H0v2h3.14c.45 1.72 2 3 3.86 3 1.86 0 3.41-1.28 3.86-3H14V7h-3.14zM7 10.2c-1.22 0-2.2-.98-2.2-2.2 0-1.22.98-2.2 2.2-2.2 1.22 0 2.2.98 2.2 2.2 0 1.22-.98 2.2-2.2 2.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/github/docs/commit/8e25c7666ce5b9576d6aa1e982a58285f5f90ef9" target="_blank" rel="noopener nofollow ugc">Merge pull request #13712 from github/repo-sync</a>
    </h4>

    <div class="github-info">
      <div class="date">
        committed <span class="discourse-local-date" data-format="ll" data-date="2022-01-04" data-time="19:28:53" data-timezone="UTC">07:28PM - 04 Jan 22 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/Octomerger" target="_blank" rel="noopener nofollow ugc">
          <img alt="Octomerger" src="https://user-images.githubusercontent.com/2119212/181071368-d9e759d5-20d8-4626-902b-d4cca65b33c0.jpeg" class="onebox-avatar-inline" width="20" height="20">
          Octomerger
        </a>
      </div>

      <div class="lines" title="changed 0 files with 0 additions and 0 deletions">
        <a href="https://github.com/github/docs/commit/8e25c7666ce5b9576d6aa1e982a58285f5f90ef9" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+0</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">repo sync</p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/f/bf623eee6ba799b530412a1359f4e46032c829d3.png" data-download-href="/uploads/short-url/rj3GD0gqG7fEqem63yu4qIl9x5h.png?dl=1" title="Screenshot_20220105-105133" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2119212/181071369-cad9a36a-1629-4575-91f6-7299182e8e56.png" alt="Screenshot_20220105-105133" data-base62-sha1="rj3GD0gqG7fEqem63yu4qIl9x5h" width="690" height="170" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/f/bf623eee6ba799b530412a1359f4e46032c829d3_2_690x170.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/f/bf623eee6ba799b530412a1359f4e46032c829d3_2_1035x255.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/f/bf623eee6ba799b530412a1359f4e46032c829d3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/f/bf623eee6ba799b530412a1359f4e46032c829d3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot_20220105-105133</span><span class="informations">1080×267 53.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>It’s doable. There are magical APIs for it</p>

**Comment by subodh2711** [comment]
<p>Actually i have a use case where we wanted to capture all the PR’s went in a release.<br>
Example :<br>
i have a repo A where i created a PR named PR-1 which is now merged to master branch of repo A and there is a target repo B where release happens , now from master branch of A changes are merged to master of B repo and after some day there is a release in B say after 5 days on that day i need to find all the PR’s went to this release. All the intial PR’s like (PR-1).</p>

**Comment by subodh2711** [comment]
<p>can we add custom fields in Pull Request?</p>

**Comment by jsoref** [comment]
<p>Not really. You can use labels to add very limited metadata.</p>
<p>You can also stick stuff into the description in a structured format.</p>
<pre><code class="lang-auto">&lt;!--
data:{"key":"value"}
--&gt;
</code></pre>
<p>And then retrieve it.</p>

**Comment by subodh2711** [comment]
<p>Great can we propagate the data fields if our PR is merged and propagated to another repo.</p>

**Comment by jsoref** [comment]
<p>Normally if you want something to propagate, you’d stick it into a commit, either in git metadata, git commit message, or normal files…</p>
